### PR TITLE
Feature/interface2

### PR DESCRIPTION
**Pull request by Tyluur**

### CODE REVIEW

File: engine/src/main/kotlin/rs/dusk/engine/model/entity/index/player/data/interfaces/InterfaceRepository.kt [hunk 1]
```diff
@@ -9,80 +9,46 @@ import com.google.common.graph.GraphBuilder
...
+        // store the interface as a child of the parent
+        if (parentId == null) {
+            parentNode!!.add(ArrayMultiTreeNode(interfaceId))
+        } else {
+            val parentNode = parentNode!!.find(parentId)
+                ?: throw IllegalStateException("Unable to find the parent node of an interface")
+            parentNode.add(ArrayMultiTreeNode(interfaceId))
         }
-        return interfaceNode
     }
```
Comment (GregHib) [inline engine/src/main/kotlin/rs/dusk/engine/model/entity/index/player/data/interfaces/InterfaceRepository.kt:46]: Would be good to get a `isInterfaceOpen` in here at some point too

**Review by GregHib — APPROVED** [review]
Don't really know what's going on here, it'll hopefully make more sense when I review /interfaces

**Comment by Tyluur** [comment]
> Don't really know what's going on here, it'll hopefully make more sense when I review /interfaces

Can you be more specific?

**Comment by GregHib** [comment]
> Can you be more specific?

I can't comment on the individual changes because I have no idea what the system in `feature/interfaces` is like in the first place, I'll give constructive feedback on the whole thing once `feature/interfaces` is complete and has a PR

**Pull request closed without merging**